Core/GDCore/Project/BehaviorConfigurationContainer.h

```diff
@@ -0,0 +1,155 @@
+/*
+ * GDevelop Core
+ * Copyright 2008-2016 Florian Rival (Florian.Rival@gmail.com). All rights
+ * reserved. This project is released under the MIT License.
+ */
+#ifndef GDCORE_BEHAVIORCONFIGURATIONCONTAINER_H
+#define GDCORE_BEHAVIORCONFIGURATIONCONTAINER_H
+
+#include <map>
+#include <memory>
+#include "GDCore/Serialization/Serializer.h"
+#include "GDCore/String.h"
+
+namespace gd {
+class PropertyDescriptor;
+class SerializerElement;
+class Project;
+class Layout;
+}  // namespace gd
+
+namespace gd {
+
+/**
+ * \brief Base class for containers of behavior configuration.
+ * They can be attached to objects (Behavior) or layouts (BehaviorsSharedData).
```

Comment:
Thank you for this explanation, very useful when reading the code and trying to understand how this is used.